packages/accounts-controller/src/AccountsController.ts

```diff
@@ -429,6 +435,29 @@ export class AccountsController extends BaseController<
     });
   }
 
+  /**
+   * Updates the metadata of the account with the given ID.
+   *
+   * @param accountId - The ID of the account for which the metadata will be updated.
+   * @param metadata - The new metadata for the account.
+   */
+  updateAccountMetadata(
+    accountId: string,
+    metadata: Partial<InternalAccount['metadata']>,
+  ): void {
+    const account = this.getAccountExpect(accountId);
+
+    this.update((currentState: Draft<AccountsControllerState>) => {
+      const internalAccount = {
+        ...account,
+        metadata: { ...account.metadata, ...metadata },
+      };
+      // Do not remove this comment - This error is flaky: Comment out or restore the `ts-expect-error` directive below as needed.
+      // // @ts-expect-error Known issue - `Json` causes recursive error in immer `Draft`/`WritableDraft` types
```

Comment:
I think this is being tracked here, so maybe we can add a link to the issue:

```suggestion
      // See: https://github.com/MetaMask/utils/issues/168
      // // @ts-expect-error Known issue - `Json` causes recursive error in immer `Draft`/`WritableDraft` types
```